### PR TITLE
[INLONG-8108][Manager] WorkflowApprover API Permissions Optimization

### LINKED ISSUE #8108: [Bug][Manager] WorkflowApprover API Permissions Optimization
### What happened

WorkflowApprover API can be accessed with too many permissions

### What you expected to happen

general user can delete,update the Process which only admin can operate

### How to reproduce

general user can delete,update the Process  operate error

### Environment

_No response_

### InLong version

master

### InLong Component

InLong Manager

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by hnrainll**
### Prepare a Pull Request

- Fixes #8108 

### Motivation

WorkflowApprover API  can be accessed with too many permissions

### Modifications

WorkflowApproverController API Permissions

### CODE REVIEW

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/WorkflowApproverController.java [hunk 1]
```diff
@@ -54,19 +56,22 @@ public class WorkflowApproverController {
...
     @ApiOperation(value = "Get approver by ID")
     @ApiImplicitParam(name = "id", value = "Workflow approver ID", dataTypeClass = Integer.class, required = true)
+    @RequiresRoles(value = UserRoleCode.ADMIN)
     public Response<ApproverResponse> get(@PathVariable Integer id) {
         return Response.success(workflowApproverService.get(id));
     }
 
     @GetMapping("/workflow/approver/list")
     @ApiOperation(value = "List workflow approvers")
+    @RequiresRoles(value = UserRoleCode.ADMIN)
```
Comment (fuweng11) [inline inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/WorkflowApproverController.java:74]: The admin check of the `/workflow/approver/list` interface cannot be added here.

Comment (hnrainll) [reply inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/WorkflowApproverController.java]: fix it

**Review by healchow — APPROVED** [review]

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/WorkflowApproverServiceImpl.java [hunk 2]
```diff
@@ -89,11 +92,18 @@ public Integer save(ApproverRequest request, String operator) {
...
+
         WorkflowApproverEntity approverEntity = approverMapper.selectById(id);
         if (approverEntity == null) {
             LOGGER.error("workflow approver not found by id={}", id);
             throw new BusinessException(ErrorCodeEnum.WORKFLOW_APPROVER_NOT_FOUND);
         }
+
+        UserInfo userInfo = LoginUserUtils.getLoginUser();
+        boolean isAdmin = userInfo.getRoles().contains(UserTypeEnum.ADMIN.name());
+        Preconditions.expectTrue(isAdmin || approverEntity.getApprovers().contains(userInfo.getName()),
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/WorkflowApproverServiceImpl.java:104]: Suggest use `userService.checkUser(approverEntity.getApprovers(), userInfo.getName(), "Current user does not have permission to get this workflow approver info")`

**Review by fuweng11 — APPROVED** [review]

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/WorkflowApproverServiceImpl.java (hunk 2 above)
Comment (hnrainll) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/WorkflowApproverServiceImpl.java]: fix it

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**